batcher/aligned-batcher/src/lib.rs

```diff
@@ -267,13 +267,18 @@ impl Batcher {
             .map_err(|e| BatcherError::TcpListenerError(e.to_string()))?;
         info!("Listening on: {}", address);
 
-        // Let's spawn the handling of each connection in a separate task.
-        while let Ok((stream, addr)) = listener.accept().await {
-            self.metrics.open_connections.inc();
-            let batcher = self.clone();
-            tokio::spawn(batcher.handle_connection(stream, addr));
+        loop {
+            match listener.accept().await {
+                Ok((stream, addr)) => {
+                    self.metrics.open_connections.inc();
+                    let batcher = self.clone();
+                    // Let's spawn the handling of each connection in a separate task.
+                    tokio::spawn(batcher.handle_connection(stream, addr));
+                    self.metrics.open_connections.dec();
```

Comment:
Note this will lead to lying metrics. The `tokio::spawn` call is not being waited for, so the next line runs concurrently with the actual connection handling, indicating it finished before that happens.

Reply:
Oh, right! Thank you, I will leave it as before.

Reply:
Addressed in [664fa44](https://github.com/yetanotherco/aligned_layer/pull/1485/commits/664fa44f2013b688567b5ab5c53dbe37ba0a8f54)